Packs/FeedNVDv2/Integrations/FeedNVDv2/FeedNVDv2.py

```diff
@@ -0,0 +1,508 @@
+
+    except KeyError:
+        try:
+            metrics = metrics["cvssMetricV30"][0]
+            return metrics["cvssData"]["version"], metrics["cvssData"]["baseScore"]
+
+        except KeyError:
+            try:
+                metrics = metrics["cvssMetricV2"][0]
+                return metrics["cvssData"]["version"], metrics["cvssData"]["baseScore"]
+
+            except KeyError:
+                return '', ''
+
+
+def test_module(client: Client):
+    """
+    Performs a simple API call to the NVD endpoint
+
+    Args:
+        client: An instance of the BaseClient connection class
+        params: A dictionary containing HTTP parameters
+
+    Returns:
+        'ok' if a successful HTTP 200 message is returned
+
+    """
+    try:
+        interval = parse_date_range('1 day', DATE_FORMAT)
+        parse_date_range(client.first_fetch, DATE_FORMAT)
+        client.get_cves("/rest/json/cves/2.0/", params={'pubStartDate': interval[0], 'pubEndDate': interval[1]})
+        return_results('ok')
+
+    except Exception as e:  # pylint: disable=broad-except
+        return_error("Invalid API key specified in integration instance configuration"
+                     + "\nError Message: " + str(e))
+
+
+def retrieve_cves(client, start_date: Any, end_date: Any, publish_date: bool):
+    """
+    Iteratively retrieves CVEs from NVD from the specified modification date
+    through the date the fetch-indicators or nvd-get-indicators command is
+    called
+
+    Args:
+        client: An instance of the BaseClient connection class
+
+    Returns:
+        Total number of CVE indicators fetched
+
+    """
+    url_suffix = "/rest/json/cves/2.0/"
+    results_per_page = 2000
+    param: dict[str, str | int] = {'startIndex': 0, 'resultsPerPage': results_per_page, 'noRejected': ''}
+    raw_cves = []  # type: ignore
+    more_to_process = True
+
+    if publish_date:
+        param['pubStartDate'] = start_date.strftime(DATE_FORMAT)
+        param['pubEndDate'] = end_date.strftime(DATE_FORMAT)
+
+    else:
+        param['lastModStartDate'] = start_date.strftime(DATE_FORMAT)
+        param['lastModEndDate'] = end_date.strftime(DATE_FORMAT)
+
+    if client.has_kev:
+        url_suffix += '&hasKev'
+
+    if client.keyword_search:
+        param['keywordSearch'] = client.keyword_search
+
+    # Collect all the indicators together
+    while more_to_process:
+        try:
+            res = client.get_cves(url_suffix, param)
+            total_results = res.get('totalResults', 0)
+
+            if total_results:
+                raw_cves += res.get('vulnerabilities')
+
+                param['startIndex'] += int(results_per_page)  # type: ignore
+
+            if (param['startIndex'] >= total_results):
+                more_to_process = False
+
+        except Exception as e:  # pylint: disable=broad-except
+            demisto.debug(e)
+        finally:
+            time.sleep(.5)
```

Comment:
we usually do not allow sleep in integration as it might cause unexpected behaviour. Is there another way to use here? Maybe just in cases of limit exceeding? and lets log it anyway.